### DISCUSSION TITLE
Commands on switches and progressbar

### DISCUSSION THREAD

**Discussion by Valenterie** [TomSchimansky/CustomTkinter#204] [General]
Hey ! Quick question for my project, i'm trying to get the value of the switches or a progressbar, but how can we get that value ?
Defining a command for the swicthes to do a certain protocol don't work since i can't just say "If you're on this side, do this, if not, do this". So how can we do it ? it works for buttons since they have only two states, pressed and not pressed, but for 3+ states the problem seems more complex.
If anyone can help me with that issue, it'll be a great ! 
Sorry if my english isn't perfect, just a french guy trying stuff lmao

**Comment by felipetesc** [comment]
If you can provide a sample code, I can help

**Reply by Valenterie** [reply]
Welcome back ! Found the issue.
In the initial code for the CTk switch, ther's a function called "self.check_state", quite a neaky one, but that's the one to use.
Here's my example : 
           
![code](https://user-images.githubusercontent.com/87931771/175833920-e4af1f19-2e40-40b9-97a4-33b909cce754.png)

 if (switch_1.check_state):  <-- That value was the thing I was trying to get

Hope this can help others !

**Reply by Valenterie** [reply]
Don't get confused by the indentation, switch_a is defined externally to the "bswitch" function.

**Reply by Valenterie** [reply]
By the way ! 
Just found the slider issue, here's my code : 

![code](https://user-images.githubusercontent.com/87931771/175836196-ef395877-e240-40c4-aba9-87b6a4eff261.png)

The thing to look for here is "sliderval = slider_1.output_value". This is the long type value of the slider (0.000... to 1.000... since it goes to that)

**I've done quite a stupid discovery**, instead of the complex work just to put the max value to 300, you can change the self.to value when initialising your slider lmao
